engine/components/renderables/Renderable.hpp

```diff
@@ -6,18 +6,22 @@
 #define BLOCKYENGINE_RENDERABLE_HPP
 
 #include "components/Component.hpp"
+#include "glm/vec4.hpp"
 
 enum RenderableType {
 	RECTANGLE,
 	ELLIPSE,
-	SPRITE
+	SPRITE,
+	ANIMATED
 };
 
 class Renderable : public Component {
 	public:
 		Renderable(GameObject& gameObject, const char* tag, RenderableType renderableType);
 		~Renderable() override;
 
+		[[nodiscard]] virtual const glm::vec4* GetSourceRect() const;
```

Comment:
Zou ik niet in Renderable doen, rect, ellipse en sprite (zie andere comment voor meer over SpriteRenderable) hebben hier niks mee te maken. zou deze method (non virtual) in AnimationRenderable zetten. AnimationController weet toch al dat het specifiek een AnimationRenderable is, dus er is geen rede om dit op het hoge niveau te zetten